### PR TITLE
Fix log popup placement [JENKINS-33109]

### PR DESCRIPTION
**Pull request by svanoort**
This fixes the log window popup placement from [JENKINS-33109](https://issues.jenkins-ci.org/browse/JENKINS-33109) and makes the window placement more natural when users have scrolled down a list of stages.

This places the popup for logs centered at the top of the visible area (a natural placement).  It also adds information about the element triggering the popup (which can be used to modify placement if desired). 

@reviewbybees

### CODE REVIEW

File: ui/src/main/js/view/stage-logs.js [hunk 1]
```diff
@@ -44,6 +44,8 @@ exports.render = function (stageDescription, onElement) {
     onElement.on(clickNSEvent, function() {
         dialog.show('Stage Logs (' + stageDescription.name + ')', stageLogsDom, {
             classes: 'cbwf-stage-logs-dialog',
+            placement: 'over-element',
+            'onElement': onElement,
```
Comment (amuniz) [inline ui/src/main/js/view/stage-logs.js]: Why `'onElement':...` and not `onElement: ...`?

File: ui/src/main/js/view/widgets/dialog/index.js [hunk 2]
```diff
@@ -59,7 +59,14 @@ exports.show = function(title, body, options) {
     bodyEl.append(body);
 
     options.modal = true;
-    var popover = popoverWidget.newPopover(title, dialog, undefined, options);
+    // Undefined for the onElement makes it in the window center
+    // WE NEED an onElement, but right now this fails
+    if (options.onElement) {
+        var popover = popoverWidget.newPopover(title, dialog, options.onElement, options);
+    } else {
+        var popover = popoverWidget.newPopover(title, dialog, undefined, options);
```
Comment (amuniz) [inline ui/src/main/js/view/widgets/dialog/index.js]: Is this conditional required? I mean, when `options.onElement` is `undefined`, the resulting call is just the one on _else_ branch. Probably `var popover = popoverWidget.newPopover(title, dialog, options.onElement, options);` is enough (without the conditional logic).

File: ui/src/main/js/view/stage-logs.js (hunk 1 above)
Comment (svanoort) [inline ui/src/main/js/view/stage-logs.js]: I was trying to be cautious but it wasn't needed, so... fixed.

File: ui/src/main/js/view/widgets/dialog/index.js (hunk 2 above)
Comment (svanoort) [inline ui/src/main/js/view/widgets/dialog/index.js]: Excess caution.. you are correct, it is not needed, so I fixed this.

File: ui/src/main/js/view/widgets/popover/index.js [hunk 3]
```diff
@@ -318,7 +318,7 @@ Popover.prototype.applyPlacement = function() {
             'top': onElementOffset.top,
             'left': onElementOffset.left + thisPopover.onElement.width() + 5
         });
-    } else if (placement === 'left') {
+    }  else if (placement === 'left') {
```
Comment (amuniz) [inline ui/src/main/js/view/widgets/popover/index.js]: Wrong spacing.

**Comment by amuniz** [comment]
:bee:

**Comment by amuniz** [comment]
super-big :bee: :)

**Comment by kwhetstone** [comment]
:bee:

**Comment by svanoort** [comment]
@reviewbybees done

File: ui/src/test/js/util/formatters-spec.js [hunk 4]
```diff
@@ -17,9 +17,9 @@ describe("util/formatters-spec", function () {
         expect(formatters.memory(1)).toBe('1B');
         expect(formatters.memory(999)).toBe('999B');
         expect(formatters.memory(1999)).toBe('1.95KB');
-        expect(formatters.memory(1999999)).toBe('1.95MB');
-        expect(formatters.memory(1999999999)).toBe('1.95GB');
-        expect(formatters.memory(1999999999999)).toBe('1.95TB');
+        expect(formatters.memory(1999999)).toBe('1.91MB');
+        expect(formatters.memory(1999999999)).toBe('1.86GB');
+        expect(formatters.memory(1999999999999)).toBe('1.82TB');
```
Comment (jglick) [inline ui/src/test/js/util/formatters-spec.js]: I guess this is unrelated, just a side effect of some changes in Node?

Comment (jglick) [inline ui/src/test/js/util/formatters-spec.js]: (I saw these test failures in `master`.)

Comment (svanoort) [inline ui/src/test/js/util/formatters-spec.js]: It's related to https://github.com/jenkinsci/pipeline-stage-view-plugin/pull/10 - for some reason the tests weren't running correctly and didn't catch the fact that we had a hard-coded dependency on (incorrect) conversion int he test specs.  This commit fixes it.
